File: Python/Conexion segura MySQL/connection.py
import mysql.connector
from cryptography.fernet import Fernet
from key import encrypted_password

# Abrir archivo donde está la clave
with open("key.key", "rb") as key_file:
    key = key_file.read()

# Cargar la contraseña encriptada desde el archivo
with open("encrypted_password.bin", "rb") as password_file:
    encrypted_password = password_file.read()

# Desencriptar la contraseña
cipher_suite = Fernet(key)
decrypted_password = cipher_suite.decrypt(encrypted_password_).decode("utf-8")

# Realizar la conexión a MySQL
connection = mysql.connector.connect(
    host="localhost",
    user="root",
    password=decrypted_password,
    database="storedb"
)

# Verificar la conexión
if connection.is_connected():
    print("Conexión exitosa a la base de datos")
else:
    print("Error en la conexión a la base de datos")

# Crear cursor
cursor = connection.cursor()

# Realizar una consulta
query = "SELECT * FROM users;"
cursor.execute(query)
result = cursor.fetchall()

# Imprimir lo datos en pantalla
print(result)


# Cerrar la conexión
connection.close()

File: C++/Sistema de Gestión de Biblioteca/cpp files/connection.cpp
#include<iostream>
#include<mysql.h>
#include<mysqld_error.h>
using namespace std;

class Connection{
private:
	char HOST[20] = "localhost";
	char USER[20] = "root";
	char PASSWORD[20] = "Ori31525588$$.";
	char DATABASE[20] = "biblioteca_db";
	int PORT = 3306;
	MYSQL* conn;

public:
	MYSQL* make_connection(){	
		conn = mysql_init(0);
		MYSQL* connected;
		connected = (mysql_real_connect(conn, HOST, USER, PASSWORD, DATABASE, PORT, NULL, 0))==nullptr ? nullptr:conn;
		return connected;
	}
};

